include/zephyr/llext/loader.h

```diff
@@ -71,6 +71,8 @@ struct llext_loader {
 	/** @cond ignore */
 	elf_ehdr_t hdr;
 	elf_shdr_t sects[LLEXT_MEM_COUNT];
```

Comment:
Is this still needed given we allocate/reference the section headers below? Should we use pointers here instead?

Reply:
Those are still useful since they host the "merged" sections, and must be writable. I think it _could_ be possible to use pointers in specific cases, but you would be modifying the original values which are used by the linking later. 
I would leave this for a further optimization.